frontend/src/components/common/date-input/date-input.tsx

```diff
@@ -1,27 +1,49 @@
-import clsx from "clsx";
-import { DateChangeEvent, DatePayload } from "common/types";
 import React from "react";
+import { useController, Control, UseControllerOptions, FieldValues, FieldErrors } from "react-hook-form";
+import { ErrorMessage } from '@hookform/error-message';
 import DatePicker from "react-datepicker";
+import clsx from "clsx";
+import { InputColor } from 'common/enums';
+
 import "react-datepicker/dist/react-datepicker.min.css";
 import "./datepicker-custom-styles.scss";
 import styles from './styles.module.scss';
 
 
 interface Props {
-  onChange: DateChangeEvent;
-  selectedDate: DatePayload;
+  name: string;
+  label: string;
+  labelHidden?: boolean;
+  placeholder?: string;
+  color: InputColor;
+  inline?: boolean;
   isDisabled?: boolean;
-  hasError?: boolean;
+  control: Control;
+  errors?: FieldErrors<FieldValues>
+  defaultValue?: UseControllerOptions<FieldValues>;
```

Comment:
```suggestion
  errors?: FieldErrors<FieldValues>
  defaultValue?: UseControllerOptions<FieldValues>;
```
It repeating in several packages. Move it to the `common` folder, and import it from there